bigtable-dataflow-parent/bigtable-hbase-beam/src/main/java/com/google/cloud/bigtable/beam/CloudBigtableScanConfiguration.java

```diff
@@ -164,17 +229,67 @@ public CloudBigtableScanConfiguration build() {
           scan = new Scan();
         }
         ReadRowsRequest.Builder builder = Adapters.SCAN_ADAPTER.adapt(scan, readHooks);
-        request = readHooks.applyPreSendHook(builder.build());
+        request =  StaticValueProvider.of(readHooks.applyPreSendHook(builder.build()));
       }
       return new CloudBigtableScanConfiguration(projectId, instanceId, tableId,
           request, additionalConfiguration);
     }
   }
 
-  // 'request' needs to be a runtime parameter because it depends on runtime parameters: project ID
-  // and instance ID.
   private final ValueProvider<ReadRowsRequest> request;
 
+  /**
+   * Provides an updated request by setting the table name in the existing request if the table name
+   * wasn't set.
+   */
+  private static class RequestWithTableNameValueProvider
```

Comment:
We talkeda bit about this offline, but I really dislike this class name. I think just RequestValueProvider makes more sense.

Reply:
In generation, I would like class name (or any name)  to be more descriptive. I don't mind changing this name to "RequestValueProvider" in this case since it is a private class. Done.